mlrun/api/db/sqldb/db.py

```diff
@@ -2209,7 +2228,9 @@ def _filter_artifacts_by_kinds(
         return filtered_artifacts
 
     def _find_functions(self, session, name, project, uids=None, labels=None):
-        query = self._query(session, Function, name=name, project=project)
```

Comment:
Note that this was previously a bug - this method only looks for exact name match. However, when it's called as part of `list_functions`, if a `tag` was provided then `_resolve_class_tag_uids` would be called which does a like query on the name, and then this method is called with an exact match. 
Doesn't look like someone really ran into this issue, but I've fixed it anyway.